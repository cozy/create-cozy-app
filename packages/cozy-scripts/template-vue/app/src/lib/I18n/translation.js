import Polyglot from 'node-polyglot'
import { DEFAULT_LANG } from './'

export let _polyglot

export const initTranslation = (
  lang,
  dictRequire,
  defaultLang = DEFAULT_LANG
) => {
  _polyglot = new Polyglot({
    phrases: dictRequire(defaultLang),
    locale: defaultLang
  })

  // Load global locales
  if (lang && lang !== defaultLang) {
    try {
      const dict = dictRequire(lang)
      _polyglot.extend(dict)
      _polyglot.locale(lang)
    } catch (e) {
      // console.warn(`The dict phrases for "${lang}" can't be loaded`)
    }
  }

  return _polyglot.t.bind(_polyglot)
}
