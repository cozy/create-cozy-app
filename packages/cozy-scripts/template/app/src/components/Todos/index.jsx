import React from 'react'

import { useQuery, isQueryLoading } from 'cozy-client'
import Spinner from 'cozy-ui/transpiled/react/Spinner'

import { getAllTodos } from '../../utils/queries'
import TodoAdd from './TodoAdd/TodoAdd'
import TodosList from './TodoList/TodosList'

export const Todos = () => {
  const { data, ...rest } = useQuery(
    getAllTodos.definition,
    getAllTodos.options
  )

  return (
    <div className="todos">
      {isQueryLoading(rest) ? (
        <Spinner size="xxlarge" middle />
      ) : (
        <div>
          <TodosList todos={data} />
          <TodoAdd />
        </div>
      )}
    </div>
  )
}

export default Todos
