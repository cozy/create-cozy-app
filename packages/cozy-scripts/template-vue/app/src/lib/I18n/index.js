/**
vue plugin that provides an I18n helper using a Higher Order Component.
*/

import { initTranslation } from './translation.js'
import { initFormat } from './format.js'

export const DEFAULT_LANG = 'en'

export function getI18nMixin(lang, dictRequire, defaultLang) {
  return {
    data: function() {
      return { lang }
    },
    methods: {
      t: initTranslation(lang, dictRequire, defaultLang),
      f: initFormat(lang, defaultLang)
    }
  }
}

export default getI18nMixin
