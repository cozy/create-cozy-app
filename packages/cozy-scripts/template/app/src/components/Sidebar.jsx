import React from 'react'
import { translate } from 'cozy-ui/react/I18n'
import { NavLink as RRNavLink } from 'react-router-dom'

import UISidebar from 'cozy-ui/transpiled/react/Sidebar'
import Nav, {
  NavItem,
  NavIcon,
  NavText,
  genNavLink
} from 'cozy-ui/transpiled/react/Nav'

import navIcon from '../assets/icons/icon-bullet-point.svg'

const NavLink = genNavLink(RRNavLink)

export const Sidebar = ({ t }) => (
  <UISidebar>
    <Nav>
      <NavItem>
        <NavLink to="/todos">
          <NavIcon icon={navIcon} />
          <NavText>{t('Nav.todos')}</NavText>
        </NavLink>
      </NavItem>
      <NavItem>
        <NavLink to="/viewhello1">
          <NavIcon icon={navIcon} />
          <NavText>{t('Nav.hello_nav_2')}</NavText>
        </NavLink>
      </NavItem>
      <NavItem>
        <NavLink to="/viewhello2">
          <NavIcon icon={navIcon} />
          <NavText>{t('Nav.hello_nav_3')}</NavText>
        </NavLink>
      </NavItem>
    </Nav>
  </UISidebar>
)

// translate() provide t() to use translations (ex: locales/en.json)
export default translate()(Sidebar)
