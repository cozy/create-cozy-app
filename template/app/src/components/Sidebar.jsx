import React from 'react'
import { NavLink as RouterLink } from 'react-router-dom'

import { useI18n } from 'cozy-ui/transpiled/react/I18n'
import UISidebar from 'cozy-ui/transpiled/react/Sidebar'
import Nav, {
  NavItem,
  NavIcon,
  NavText,
  NavLink
} from 'cozy-ui/transpiled/react/Nav'

import BulletPoint from 'src/assets/icons/icon-bullet-point.svg'

const Sidebar = () => {
  const { t } = useI18n()

  return (
    <UISidebar>
      <Nav>
        <NavItem>
          <RouterLink
            to="/todos"
            className={NavLink.className}
            activeClassName={NavLink.activeClassName}
          >
            <NavIcon icon={BulletPoint} />
            <NavText>{t('nav.todos')}</NavText>
          </RouterLink>
        </NavItem>
        <NavItem>
          <RouterLink
            to="/viewhello1"
            className={NavLink.className}
            activeClassName={NavLink.activeClassName}
          >
            <NavIcon icon={BulletPoint} />
            <NavText>{t('nav.hello_nav_2')}</NavText>
          </RouterLink>
        </NavItem>
        <NavItem>
          <RouterLink
            to="/viewhello2"
            className={NavLink.className}
            activeClassName={NavLink.activeClassName}
          >
            <NavIcon icon={BulletPoint} />
            <NavText>{t('nav.hello_nav_3')}</NavText>
          </RouterLink>
        </NavItem>
      </Nav>
    </UISidebar>
  )
}

export default Sidebar
