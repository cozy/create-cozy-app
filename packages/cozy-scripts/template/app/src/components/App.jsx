import React from 'react'
import { Route, Switch, Redirect, HashRouter } from 'react-router-dom'
import { hot } from 'react-hot-loader'

import { Layout, Main, Content } from 'cozy-ui/transpiled/react/Layout'
import { Sprite as IconSprite } from 'cozy-ui/transpiled/react/Icon'

import Sidebar from './Sidebar'
import Todos from './Todos'
import Hello1 from './HelloViews/Hello1'
import Hello2 from './HelloViews/Hello2'

const App = () => (
  <HashRouter>
    <Layout>
      <Sidebar />
      <Main>
        <Content className="app-content">
          <Switch>
            <Route path="/todos" component={Todos} />
            <Route path="/viewhello1" component={Hello1} />
            <Route path="/viewhello2" component={Hello2} />
            <Redirect from="/" to="/todos" />
            <Redirect from="*" to="/todos" />
          </Switch>
        </Content>
      </Main>
      <IconSprite />
    </Layout>
  </HashRouter>
)

/*
  Enable Hot Module Reload using `react-hot-loader` here
  We enable it here since App is the main root component
  No need to use it anywhere else, it sould work for all
  child components
*/
export default hot(module)(App)
