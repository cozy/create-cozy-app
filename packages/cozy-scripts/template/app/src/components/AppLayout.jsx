/* global cozy */
import React from 'react'
import { Outlet } from 'react-router-dom'
import cx from 'classnames'

import { Layout, Main, Content } from 'cozy-ui/transpiled/react/Layout'
import Sprite from 'cozy-ui/transpiled/react/Icon/Sprite'
import Alerter from 'cozy-ui/transpiled/react/Alerter'
import { useI18n } from 'cozy-ui/transpiled/react/I18n'
import useBreakpoints from 'cozy-ui/transpiled/react/hooks/useBreakpoints'
import MuiCozyTheme from 'cozy-ui/transpiled/react/MuiCozyTheme'
import BarTitle from 'cozy-ui/transpiled/react/BarTitle'
import { useClient } from 'cozy-client'

import Sidebar from 'src/components/Sidebar'

const AppLayout = () => {
  const { t } = useI18n()
  const client = useClient()
  const { isMobile } = useBreakpoints()
  const { BarCenter } = cozy.bar

  return (
    <Layout>
      {isMobile && (
        <BarCenter>
          <MuiCozyTheme>
            <BarTitle>{client.appMetadata.slug}</BarTitle>
          </MuiCozyTheme>
        </BarCenter>
      )}
      <Sidebar />
      <Main>
        <Content
          className={cx({
            'u-mh-half u-mv-1': isMobile,
            'u-mh-2 u-mv-1': !isMobile
          })}
        >
          <Outlet />
        </Content>
      </Main>
      <Alerter t={t} />
      <Sprite />
    </Layout>
  )
}

export default AppLayout
