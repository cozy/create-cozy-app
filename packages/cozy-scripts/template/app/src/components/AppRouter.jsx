import React from 'react'
import {
  Route,
  Navigate,
  RouterProvider,
  createHashRouter,
  createRoutesFromElements
} from 'react-router-dom'

import TodoWrapper from 'src/components/Todos/TodoWrapper'
import Hello1 from 'src/components/HelloViews/Hello1'
import Hello2 from 'src/components/HelloViews/Hello2'
import AppLayout from 'src/components/AppLayout'

const AppRouter = () => {
  const routes = (
    <Route path="/" element={<AppLayout />}>
      <Route path="todos" element={<TodoWrapper />} />
      <Route path="viewhello1" element={<Hello1 />} />
      <Route path="viewhello2" element={<Hello2 />} />
      <Route path="*" element={<Navigate to="/todos" replace />} />
    </Route>
  )
  const router = createHashRouter(createRoutesFromElements(routes))

  return <RouterProvider router={router} />
}

export default AppRouter
