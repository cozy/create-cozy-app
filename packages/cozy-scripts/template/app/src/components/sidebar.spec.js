'use strict'

/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'

import { mockT } from '../lib/I18n'
import { Sidebar } from 'components/Sidebar'

describe('Sidebar component', () => {
  it('should be rendered correctly', () => {
    const component = shallow(<Sidebar t={mockT} />).getElement()
    expect(component).toMatchSnapshot()
  })
})
