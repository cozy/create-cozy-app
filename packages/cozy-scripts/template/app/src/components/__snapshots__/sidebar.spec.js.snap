// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar component should be rendered correctly 1`] = `
<Sidebar>
  <Nav>
    <NavItem>
      <Unknown
        to="/todos"
      >
        <NavIcon
          icon="test-file-stub"
        />
        <NavText>
          Todos
        </NavText>
      </Unknown>
    </NavItem>
    <NavItem>
      <Unknown
        to="/viewhello1"
      >
        <NavIcon
          icon="test-file-stub"
        />
        <NavText>
          Second
        </NavText>
      </Unknown>
    </NavItem>
    <NavItem>
      <Unknown
        to="/viewhello2"
      >
        <NavIcon
          icon="test-file-stub"
        />
        <NavText>
          Third
        </NavText>
      </Unknown>
    </NavItem>
  </Nav>
</Sidebar>
`;
